src/ucp/wireup/wireup_cm.c

```diff
@@ -654,51 +619,30 @@ static unsigned ucp_ep_cm_remote_disconnect_progress(void *arg)
                                           UCP_EP_FLAG_CLOSE_REQ_VALID)) {
         ucp_request_complete_send(ucp_ep_ext_control(ucp_ep)->close_req.req,
                                   UCS_OK);
-        return 1;
-    }
-
-    if (ucp_ep->flags & UCP_EP_FLAG_CLOSED) {
-        /* the ep is closed by API but close req is not valid yet (checked
-         * above), it will be set later from scheduled
-         * @ref ucp_ep_close_flushed_callback */
-        ucs_debug("ep %p: ep closed but request is not set, waiting for"
-                  " the flush callback", ucp_ep);
-        goto err;
+        return;
     }
 
     if (!(ucp_ep->flags & UCP_EP_FLAG_REMOTE_CONNECTED)) {
-        /* CM disconnect happens during WIREUP MSGs exchange phase, when EP
-         * is locally connected to the peer */
-        goto err;
-    }
-
-    /*
-     * TODO: set the ucp_ep to error state to prevent user from sending more
-     *       ops.
-     */
-    ucs_assert(ucp_ep->flags & UCP_EP_FLAG_FLUSH_STATE_VALID);
-    ucs_assert(!(ucp_ep->flags & UCP_EP_FLAG_CLOSED));
-    req = ucp_ep_flush_internal(ucp_ep, 0, &ucp_request_null_param, NULL,
```

Comment:
so remote disconnect does not generate flush anymore?

Reply:
yes, we discussed this with @evgeny-leksikov - no reason doing flush(), since a local side was disconnected from a peer
also, this code was dead and unreachable
after I've fixed unreachability by moving REMOTE_CONENCTED flag removing from EP flags, it starts failing due to recent changes in close_nbx(FORCE) that does discarding of lanes and destroys UCP EP. So, UCP EP could be touched from flush completion callback after it is destroyed.